Csg.Data/DbConvert.cs

```diff
@@ -144,10 +144,8 @@ public static DbType TypeCodeToDbType(TypeCode typeCode)
         /// <returns></returns>
         public static System.Data.DbType TypeToDbType(Type type)
         {
-            if (type == typeof(Nullable<>))
-            {
-                type = type.GetGenericArguments()[0];
-            }
+            var underlyingType = Nullable.GetUnderlyingType(type);
```

Comment:
Was there a reason you got rid of the check whether type is a Nullable<> type and only get the underlying type when it is? I guess it looks like it still works when type is not Nullable, but it might be a little confusing.

Reply:
@LeBecca yes, sorry I didn't make it clear, but that *was* the actual bug. If checkout this branch, and replace the above code with the old version, and then run my new tests, you'll see they fail on the null versions. 

As far as I can tell, checking ```type == typeof(Nullable<>)``` never worked.

Reply:
Alternatively we could branch & optimize for the non-nullable path using the method they use to implement GetNullableType.

https://github.com/microsoft/referencesource/blob/master/mscorlib/system/nullable.cs#L132

Reply:
Since this is netcore I guess the correct source link for that method is below, but more or less the same.

https://github.com/dotnet/runtime/blob/f56ef95669c115510c0b6dc294634d932fd4e49b/src/libraries/System.Private.CoreLib/src/System/Nullable.cs#L98

Reply:
Ok, I understand now. That returns null if the type is not a nullable type, which then falls through the "underlyingType ?? type" statement.